source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -210,40 +210,53 @@ namespace
+        PvtPrim* node = findPrimOrThrow(RtToken(nodeElem->getName()), parent, mapper);
+        for (InputPtr elemInput : nodeElem->getInputs())
         {
-            PvtPrim* node = findPrimOrThrow(RtToken(nodeElem->getName()), parent, mapper);
-            for (const InputPtr& elemInput : nodeElem->getInputs())
+            PvtInput* input = findInputOrThrow(RtToken(elemInput->getName()), node);
+            string connectedNodeName = elemInput->getNodeName();
+            if (connectedNodeName.empty())
             {
-                PvtInput* input = findInputOrThrow(RtToken(elemInput->getName()), node);
-                string connectedNodeName = elemInput->getNodeName();
-                if (connectedNodeName.empty())
-                {
-                    connectedNodeName = elemInput->getNodeGraphString();
-                }
-                if (!connectedNodeName.empty())
+                connectedNodeName = elemInput->getNodeGraphString();
+            }
+            if (!connectedNodeName.empty())
+            {
+                PvtPrim* connectedNode = findPrimOrThrow(RtToken(connectedNodeName), parent, mapper);
+                RtToken outputName(elemInput->getOutputString());
+                if (outputName == EMPTY_TOKEN && connectedNode)
                 {
-                    PvtPrim* connectedNode = findPrimOrThrow(RtToken(connectedNodeName), parent, mapper);
-                    RtToken outputName(elemInput->getOutputString());
-                    if (outputName == EMPTY_TOKEN && connectedNode)
+                    RtNode rtConnectedNode(connectedNode->hnd());
+                    auto output = rtConnectedNode.getOutput();
+                    if (output)
                     {
-                        RtNode rtConnectedNode(connectedNode->hnd());
-                        auto output = rtConnectedNode.getOutput();
-                        if (output)
-                        {
-                            outputName = output.getName();
-                        }
+                        outputName = output.getName();
                     }
-                    PvtOutput* output = findOutputOrThrow(outputName, connectedNode);
-
-                    createConnection(output, input, elemInput->getChannels(), stage);
                 }
+                PvtOutput* output = findOutputOrThrow(outputName, connectedNode);
+
+                createConnection(output, input, elemInput->getChannels(), stage);
             }
         }
     }
 
+    void createNodeConnections(const vector<NodePtr>& nodeElements, PvtPrim* parent, PvtStage* stage, const PvtRenamingMapper& mapper)
+    {
+        for (auto nodeElem : nodeElements)
+        {
+            createNodeConnection(nodeElem->asA<InterfaceElement>(), parent, stage, mapper);
+        }
+    }
+
+    void createNodeGraphConnections(const vector<NodeGraphPtr>& nodeElements, PvtPrim* parent, PvtStage* stage, const PvtRenamingMapper& mapper)
```

Comment:
No real change in logic above. Just make it so that we can check for connections between nodegraph and upstream nodes / nodegraphs now. Note added 2 interfaces since can't really cast from a constant array of node vs nodegraph ptrs so just allowed 2 interfaces and the per item check will cast to an interfaceElement check.